src/pages/suggestions/[suggestion].tsx

```diff
@@ -3,67 +3,95 @@ import { MusicalNoteIcon, XMarkIcon } from "@heroicons/react/24/solid"
 import Link from "next/link"
 import ProgressionBar from "@/components/suggestion/progression-bar"
 import Image from "next/image"
-import { Instruments } from "@/constants/instruments"
-import { ISuggestion } from "@/interfaces/suggestion"
-import WithProtectedRoute from "@/components/protected-route/protected-route"
-import { GetStaticPaths, GetStaticProps } from "next"
-import { getSuggestion, updateSuggestion } from "@/services/suggestion.service"
-import { useAuthContext } from "@/context/auth-context"
-import { IUser } from "@/interfaces/user";
+import { GetServerSideProps } from "next"
+import {
+  deleteDivision,
+  getInstrumentImage,
+  getSuggestion,
+  insertDivision,
+} from "@/services/suggestion.service"
+import { formatDistanceToNow } from "date-fns"
+import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs"
+import { PostgrestError } from "@supabase/supabase-js"
+import {
+  Division,
+  DivisionDatabaseOperation,
+  Suggestion,
+  SuggestionInstrument,
+} from "@/types/database-types"
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
 
 interface SuggestionProps {
-  props: ISuggestion
+  suggestion: Suggestion
+  error: PostgrestError | null
 }
 
-const Suggestion: FC<SuggestionProps> = ({ props }) => {
-  const [suggestion, setSuggestion] = useState<ISuggestion>(props)
-  const { user } = useAuthContext()
-  const username = user?.additionalUserData.username as string
-
-  const setDate = (): string => {
-    return new Date(suggestion?.date.seconds).toLocaleDateString('en-us', { month: "long", day: "numeric" })
+const SuggestionPage: FC<SuggestionProps> = (props: SuggestionProps) => {
+  const [suggestion, setSuggestion] = useState<Suggestion>(props.suggestion)
+  const [error, setError] = useState<PostgrestError | null>(props.error)
+  const user = useUser()
+  const supabase = useSupabaseClient<Database>()
+  const uid = user?.id
+
+  function updateSuggestion() {
```

Comment:
I think it would be nice for consistency to use the arrow notation for functions everywhere